### ISSUE TITLE
[java] Improve JavaAnnotationSuppressor

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#5773]
Left-over tasks from #5609:

* Properly support unused detection for private methods
* Enable currently disabled rule tests for UnnecessaryWarningSuppression
* Improve rule message when suppression overlaps (e.g. `{"PMD", "all"}`)

**Comment by ben-manes** [comment]
When I cross-compile against newer JDKs, the `CloseResource` warning occurs on 25-ea for `ExecutorService` usages. When suppressed then `UnnecessaryWarningSuppression` occurs in lower JDKs and suppressing that doesn't seem to work at the statement level. At the method level then it complained about unnecessarily suppressing itself. So some fine tuning for cross-builds would be helpful.

```java
@SuppressWarnings({"PMD.CloseResource", "PMD.UnnecessaryWarningSuppression"})
ScheduledExecutorService scheduledExecutor = Mockito.mock();
```

```
UnnecessaryWarningSuppression:  Unnecessary suppression "PMD.UnnecessaryWarningSuppression" in @SuppressWarnings annotation
```

**Comment by oowekyala** [comment]
@ben-manes I don't think we really can support cross builds. If you cross build for several JDKs then you will need to run this rule only once, and exclude it from the other runs (IIUC).

Also UnnecessaryWarningSuppression cannot be suppressed by any means (it would run into a chicken and egg problem), which is why it reports itself here.

**Comment by adangel** [comment]
@ben-manes if you use maven-pmd-plugin, you can use [Violation Exclusions](https://maven.apache.org/plugins/maven-pmd-plugin/examples/violation-exclusions.html). PMD will still report UnnecessaryWarningSuppression, but m-pmd-p then ignores it and doesn't fail the build.

**Comment by ben-manes** [comment]
For now I have the rule excluded and enable it manually to review. I might later use gradle toolchains to force a minimum version with pmd, since GHA runs it with the latest release. Currently the build would use whatever toolchain is requested (11-25) to cross build and for compatibility with analyzers that are slower to add jdk compatibility.

**Comment by ben-manes** [comment]
For the Gradle plugin, I was able to set the java toolchain on the quality plugins so it doesn't default to the `java.toolchain.languageVersion` which may be cross built, so my prior concern is no longer relevant.

```kts
tasks.withType<AbstractCodeQualityTask>().configureEach {
  javaLauncher = javaToolchains.launcherFor {
    vendor = java.toolchain.vendor
    languageVersion = javaRuntimeVersion
    implementation = java.toolchain.implementation
    nativeImageCapable = java.toolchain.nativeImageCapable
  }
}
```
